src/Recruiter/Job/Repository.php

```diff
@@ -241,6 +243,181 @@ public function recentHistory($group = null, T\Moment $at = null, array $query =
         ];
     }
 
+    /**
+     * @param int
+     */
+    public function countSlowRecentJobs(
+        T\Moment $lowerLimit,
+        T\Moment $upperLimit,
+        $secondsToConsiderJobAsSlow=5
+    )
+    {
+        $archived = $this->archived->aggregate([
+            [
+                '$match' => [
+                    'last_execution.ended_at' => [
+                        '$gte' => T\MongoDate::from($lowerLimit),
+                    ],
+                ]
+            ],
+            [
+                '$project' => [
+                    '_id' => '$_id',
+                    'execution_time' => [
+                        '$subtract' => [
+                            '$last_execution.ended_at',
+                            '$last_execution.started_at'
+                        ]
+                    ],
+                ]
+            ],
+            [
+                '$match' => [
+                    'execution_time' => [
+                        '$gt' => $secondsToConsiderJobAsSlow*1000 
+                    ]
+                ],
+            ],
+        ])['result'];
+
+        return count($archived) + $this->scheduled->count([
+            'scheduled_at' => [
+                '$gte' => T\MongoDate::from($lowerLimit),
+                '$lte' => T\MongoDate::from($upperLimit)
+            ],
+            'last_execution.started_at' => [
+                '$exists' => true,
+            ],
+            'last_execution.ended_at' => [
+                '$exists' => true,
+            ],
+        ]);
+    }
+
+    /**
+     * @param int
+     */
+    public function countRecentJobsWithManyAttempts(T\Moment $lowerLimit)
+    {
+        $archived = $this->archived->count([
+            'last_execution.ended_at' => [
+                '$gte' => T\MongoDate::from($lowerLimit),
+            ],
+            'attempts' => [
+                '$gt' => 1
+            ]
+        ]);
+        $scheduled = $this->scheduled->count([
+            'attempts' => [
+                '$gt' => 1
+            ]
+        ]);
+        return $archived+$scheduled;
+    }
+
+    /**
+     * @param int
+     */
+    public function countDelayedScheduledJobs()
+    {
+        return $this->scheduled->count([
+            'scheduled_at' => [
+                '$lte' => T\MongoDate::from(T\hour(5)->sinceNow())
+            ]
+        ]);
+    }
+
+    public function delayedScheduledJobs()
+    {
+        return $this->map(
+            $this->scheduled->find([ 
+                'scheduled_at' => [
+                    '$lte' => T\MongoDate::from(T\hour(5)->sinceNow())
```

Comment:
@silvadanilo @gabrielelana: 
as Danilo suggested a scheduled jobs really considered as "delayed" should be scheduled job scheduled some time's back in the past meaning we can't say that a job which was scheduled >= 1 hour ago is really a delayed one. Following Dani's advise I set a limit of the scheduled jobs in the past until 5 hours ago.
Do you agree with this value?

Reply:
I think that the time should be injected, so you can choose it can vary depending on the case (i.e. for generic jobs 5 hours late are a lot (so the settings is ok), while for exactor jobs 5 hour is absolutely normal)